packages/ag-charts-community/src/api/preset/chartTypeOriginator.ts

```diff
@@ -0,0 +1,34 @@
+import type { AgFinancialChartOptions, AgInitialStateChartType } from 'ag-charts-types';
+
+import type { ChartService } from '../../chart/chartService';
+import { isString } from '../../util/type-guards';
+import type { MementoOriginator } from '../state/memento';
+
+export type ChartTypeMemento = AgInitialStateChartType;
+
+export class ChartTypeOriginator implements MementoOriginator<ChartTypeMemento> {
+    public mementoOriginatorKey = 'chartType' as const;
+
+    constructor(private readonly chartService: ChartService) {}
+
+    public createMemento() {
+        let chartType = (this.chartService.publicApi?.getOptions() as AgFinancialChartOptions)?.chartType;
+        if (chartType === 'range-area') chartType = 'candlestick';
```

Comment:
Since `'range-area'` is deprecated, would we want to support it in this new api?

Reply:
Is there any programatic way to reach this state? (Guessing no?) I agree I don't think we need to support this value strictly.

Reply:
`'range-area'` has been removed